### ISSUE TITLE
Register purl pkg: scheme @ IANA

### ISSUE THREAD

**Issue by pombredanne** [package-url/purl-spec#124]
This came up as a topic in a recent discussion.... I think that with the actual adoption, now is a good time!
I will draft the submission here and start the process of registering at IANA.

**Comment by elear** [comment]
Yes please.  The registration template can be found in [RFC7595](https://www.rfc-editor.org/rfc/rfc7595.html#section-7.4) Section 7.4, but do read the rest of the document.  Today, anyone could conceivably request "pkg:" for other reasons.  If you need help, contact me.

**Comment by pombredanne** [comment]
@elear I have been working on the registration a while back... I just pushed a PR for review as #141 
I will socialize this today with the IETF URI review list too.

@coderpatros ^

**Comment by elear** [comment]
Fantastic!  I've just added purl support for my apt2sbom cyclonedx generation tool.  I'll do the same with spdx when I understand where they want that info.  Let me know if you want an extra pair of eyes.

**Comment by pombredanne** [comment]
I just sent this email to the uri-review list
https://mailarchive.ietf.org/arch/msg/uri-review/6m26SxW5e1eVzzQvQ-04oIbnqUY/

You guys may want to chime in there too.

**Comment by whiskeysierra** [comment]
(The IANA URI review highlights that it's not a URL, but a URI. But I guess it's to late for this project's name to be changed accordingly.)

**Comment by pombredanne** [comment]
@whiskeysierra re:

> (The IANA URI review highlights that it's not a URL, but a URI. But I guess it's to late for this project's name to be changed accordingly.)

Thank you for stopping by!

I am not sure I read the same from the current IANA discussion, unless I am missing something... do you have a specific quote and pointer you could share here?

FWIW, there have been several debates on this topic in the past. Do you mind to have a look at https://github.com/package-url/purl-spec/issues/5 and https://github.com/package-url/purl-spec/issues/9 for some background?

**Comment by whiskeysierra** [comment]
(Sorry for derailing this issue. Thanks a bunch for the links, those are insightful!)

**Comment by elear** [comment]
Hmm.  Purl reminds us of treasures, while puri reminds us of tasty Indian cuisine ;-)  I'd suggest not fixating on the name...

**Comment by Hritik14** [comment]
Just a note, after this is done we would want to have it on https://en.wikipedia.org/wiki/List_of_URI_schemes as well <3

**Comment by iamwillbar** [comment]
@pombredanne is there anything I can do to help get this registration submitted?

**Comment by sjn** [comment]
Apologies for waking this issue from it's coma, but are there any movements on this and #141? :-)

**Comment by elear** [comment]
Hi!  I just chatted with Ted Hardie who did the initial review.  He'd be willing to help us get across the line.  @pombredanne just ping me in email and we can set something up.

**Comment by pombredanne** [comment]
The registration email has been sent
https://mailarchive.ietf.org/arch/msg/uri-review/WIXi8IEc4O6JzzY0djZTN56k8n4/

**Comment by jkowalleck** [comment]
and people are already using it - or plan on using it - as a URI 👍 
https://github.com/CycloneDX/specification/issues/612#issuecomment-5427943649

**Comment by mjherzog** [comment]
July 27 email from David Dong at IANA:
We've added provisional URI scheme "pkg" to the registry:
- https://www.iana.org/assignments/uri-schemes/prov/pkg
- Registry: https://www.iana.org/assignments/uri-schemes

Per the designated expert for the URI Schemes registry, we're also notifying the [uri-review@ietf.org](mailto:uri-review@ietf.org) mailing list that this provisional registration has been completed.

Closing this issue for the provisional IANA registration in favor of #985 to track permanent registration.
